src/cli/MeasureUpdateCommand.cpp

```diff
@@ -367,6 +368,56 @@ namespace cli {
       fmt::print("{}\n", resultStr);
       return;
     } else if (opt.run_tests) {
+
+      auto canonicalTestDir = openstudio::filesystem::canonical(opt.directoryPath);
+
+      // Pytest
+      fmt::print(fmt::fg(fmt::color::yellow),
+                 "┌{0:─^{2}}┐\n"
+                 "│{1: ^{2}}│\n"
+                 "└{0:─^{2}}┘",
+                 "", "Starting Python Tests", 80);
+      fmt::print("\n");
+      auto pytestOutDir = canonicalTestDir / "test_results" / "pytest";
+
+      // -o junit_familar=legacy is to mimic having a pytest.ini with:
+      // [pytest]
+      // junit_family=legacy
+      // This uses xunit1 format, and adds the line + file at which error occured, something we don't have with xunit2 (default)
+
+      auto runPytestCmd = fmt::format(
+        R"python(
+import pytest
+
+pytest.main([
+  "--junit-xml={junit}",
+  "--cov={test_dir}",
+  "--cov-report=term-missing",
+  "--cov-report=json:{json}",
+  "--cov-report=lcov:{lcov}",
+  "--cov-report=html:{html}",
+  "--cov-report=xml:{xml}",
+  "-o", "junit_family=legacy",
+  "{test_dir}",
+])
+)python",
+        fmt::arg("junit", (pytestOutDir / "junit.xml").generic_string()),  //
+        fmt::arg("test_dir", canonicalTestDir.generic_string()),           //
+        fmt::arg("json", (pytestOutDir / "python_coverage.json").generic_string()),
+        fmt::arg("lcov", (pytestOutDir / "python_coverage.lcov").generic_string()),
+        fmt::arg("html", (pytestOutDir / "python_coverage_html").generic_string()),
+        fmt::arg("xml", (pytestOutDir / "python_coverage.xml").generic_string()));
+
+      fmt::print("runPytestCmd={}\n", runPytestCmd);
+      pythonEngine->exec(runPytestCmd);
+
+      fmt::print(fmt::fg(fmt::color::red),
+                 "┌{0:─^{2}}┐\n"
+                 "│{1: ^{2}}│\n"
+                 "└{0:─^{2}}┘",
+                 "", "Starting ruby tests", 80);
+      fmt::print("\n");
```

Comment:
openstudio measure --run-tests will now run Pytests with coverage

this won't be on the dashboard yet.

@DavidGoldwasser @kbenne will need to figure out who should trigger pytest, the C++ CLI or the measure tester gem.

Anyways, for now, I think like this we could just tweak measure-tester-gem to consume the pytest xmls if they are present, since the ruby code runs after the python one.

But if you use measure-tester as a gem and use the rake task, then you'd have to manually run pytest.

Reply:
This is great. We can try to do work in measure tester gem shortly after 3.8.0 is out to include in 3.8.x or 3.9.0. Many Python measures may also use additional python libraries. In the case when their is extra libraries, I wouldn't expect tests to run with out of the box install without some extra configuration for the additional libraries.

Reply:
it should work as long as you pass `--python_path` options to the CLI.